policy-test/tests/accepted_http_route_status.rs

```diff
@@ -0,0 +1,421 @@
+                proxy_protocol: Some(k8s::policy::server::ProxyProtocol::Http1),
+            },
+        };
+        let _server = create(&client, server).await;
+
+        // Create a route that references both parents.
+        let _route = create(
+            &client,
+            mk_inbound_route(&ns, "test-multiple-parents-route", Some(srv_refs)),
+        )
+        .await;
+        // Wait until route is updated with a status
+        let parent_status = await_route_status(&client, &ns, "test-multiple-parents-route")
+            .await
+            .parents;
+
+        // Find status for invalid parent and extract the condition
+        let invalid_cond = parent_status
+            .iter()
+            .find(|route_status| route_status.parent_ref.name == "test-invalid-server")
+            .expect("must have a route status set for invalid server")
+            .conditions
+            .clone()
+            .into_iter()
+            .find(|cond| cond.type_ == "Accepted")
+            .expect("must have at least one 'Accepted' condition set for invalid parent");
+        // Route shouldn't be accepted
+        assert_eq!(invalid_cond.status, "False");
+        assert_eq!(invalid_cond.reason, "NoMatchingParent");
+
+        // Find status for valid parent and extract the condition
+        let valid_cond = parent_status
+            .iter()
+            .find(|route_status| route_status.parent_ref.name == "test-valid-server")
+            .expect("must have a route status set for valid server")
+            .conditions
+            .clone()
+            .into_iter()
+            .find(|cond| cond.type_ == "Accepted")
+            .expect("must have at least one 'Accepted' condition set for valid parent");
+        assert_eq!(valid_cond.status, "True");
+        assert_eq!(valid_cond.reason, "Accepted")
+    })
+    .await
+}
+
+#[tokio::test(flavor = "current_thread")]
+async fn inbound_no_parent_ref_patch() {
+    with_temp_ns(|client, ns| async move {
+        // A route may not include any parent references. When that's the case,
+        // we expect the controller to simply ignore it.
+        let _route = create(
+            &client,
+            mk_inbound_route(&ns, "test-no-parent-refs-route", None),
+        )
+        .await;
+
+        // Status may not be set straight away. To account for that, wrap a
+        // status condition watcher in a timeout.
+        let api = kube::Api::namespaced(client.clone(), &ns);
+        let resolved = tokio::time::timeout(
+            // 5 seconds should be enough for a route to be patched
+            tokio::time::Duration::from_secs(5),
+            kube::runtime::wait::await_condition(
+                api,
+                "test-no-parent-refs-route",
+                |obj: Option<&k8s::policy::HttpRoute>| -> bool {
+                    obj.and_then(|route| route.status.as_ref()).is_some()
+                },
+            ),
+        )
+        .await;
```

Comment:
Maybe have this timeout logic embedded into `await_route_status` itself, returning an error result if timed out?

Reply:
That's a great point. I'm a bit split here... `await_route_status` already uses a timeout since it calls into `await_condition` (which is just a wrapper for the function you see above. https://github.com/linkerd/linkerd2/blob/main/policy-test/src/lib.rs#L71-L75). 

I initially wanted to edit `await_condition` to return the timeout error instead of failing fast once it's been hit, but that meant I had to edit other portions of the code. I ended up keeping it like this because I thought it was simpler (and creating in its own function seemed like too much work).

We could perhaps modify `await_route_status` to instead do all of this manually instead of relying on `await_condition` but then we probably get a bit more duplication. Anyway, given all the context, I'm happy to take a stab at this if you think it's worth resolving. Lmk.